lib/buffer.js

```diff
@@ -339,10 +339,14 @@ function slowToString(encoding, start, end) {
 
 
 Buffer.prototype.toString = function() {
-  const length = this.length | 0;
-  if (arguments.length === 0)
-    return this.utf8Slice(0, length);
-  return slowToString.apply(this, arguments);
+  if (arguments.length === 0) {
+    var result = this.utf8Slice(0, this.length);
```

Comment:
Whoops. Thought this was in a test file. Yeah it should definitely be fixed. Going to run CI right now. If everything looks good I'll fix this and land it.